engine/access/ingestion/engine.go

```diff
@@ -187,6 +187,34 @@ func New(
 	return e, nil
 }
 
+func (e *Engine) Start(parent irrecoverable.SignalerContext) {
+	rootBlock, err := e.state.Params().Root()
+	if err != nil {
+		parent.Throw(fmt.Errorf("failed to get root block: %w", err))
+	}
+
+	rootSnapshot := e.state.AtBlockID(rootBlock.ID())
+
+	isSporkRootSnapshot, err := protocol.IsSporkRootSnapshot(rootSnapshot)
+	if err != nil {
+		parent.Throw(fmt.Errorf("could not check if root snapshot is a spork root snapshot: %w", err))
+	}
+
+	// This is useful for dynamically bootstrapped access node, they will request missing collections. In order to ensure all txs
+	// from the missing collections can be verified, we must ensure they are referencing to known blocks.
+	// That's why we set the full block height to be rootHeight + TransactionExpiry, so that we only request missing collections
+	// in blocks above that height.
+	if !isSporkRootSnapshot {
+		firstFullHeight := rootBlock.Height + flow.DefaultTransactionExpiry
+		err := e.blocks.UpdateLastFullBlockHeight(firstFullHeight)
```

Comment:
We should only ever set it once. 

Would it overwrite the value on every restart?

Reply:
+1 It will overwrite on every restart, we need insert-if-not-exists here.

Reply:
https://github.com/onflow/flow-go/pull/2974/commits/f5b1e1da1be9792dabe98246a86739761d1dbfb6